### PR TITLE
[Redesign] Added Report Abuse page

### PR DESCRIPTION
**Pull request by loic-sharma**
report-abuse-one

report-abuse-two

Example URL: https://localhost/packages/jQuery/3.1.1/ReportAbuse

@microsoftsam

### CODE REVIEW

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml [hunk 1]
```diff
@@ -2,98 +2,166 @@
...
+    <h1 class="page-heading">Report Abuse</h1>
 
-<p class="message">
-    <strong>Important: Please do not use this form to report a bug in a package you are using!</strong><br />
-    This form is for reporting <em>abusive</em> packages such as
-    packages containing <em>malicious code</em> or spam. If "@Model.PackageId" simply doesn't
-    work, or if you need help getting the package installed, please
-    <a href="@Url.Action(actionName: "ContactOwners", controllerName: "Packages", routeValues: new { id = Model.PackageId })" title="contact the owners">contact the owners instead.</a>
-</p>
+    <div class="alert alert-warning">
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: role="alert"

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml [hunk 2]
```diff
@@ -2,98 +2,166 @@
...
+        <ul class="list-unstyled ms-Icon-ul">
+            <li>
+                <i class="ms-Icon ms-Icon--Warning" aria-hidden="true"></i>
```
Comment (microsoftsam) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: Consider a larger font for this icon. Looks a little small.

**Comment by microsoftsam** [comment]
Just aria-role on the alert widget. Otherwise looks good :)

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml (hunk 2 above)
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: I brought this up with Jon about this, he's fine with the current size.

File: src/NuGetGallery/Scripts/gallery/jquery.validate.js [hunk 3]
```diff
@@ -0,0 +1,1231 @@
+﻿/*!
+ * jQuery Validation Plugin 1.11.1
+ *
+ * http://bassistance.de/jquery-plugins/jquery-plugin-validation/
```
Comment (joelverhagen) [inline src/NuGetGallery/Scripts/gallery/jquery.validate.js]: Don't we have a new version of this now: **jquery.validate-1.16.0.js**?

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml [hunk 4]
```diff
@@ -2,98 +2,166 @@
...
-            @Html.LabelFor(m => m.Reason)
-            @Html.EnumDropDownListFor(m => m.Reason, Model.ReasonChoices, "<Choose a Reason>")
-            @Html.ValidationMessageFor(m => m.Reason)
-        </div>
-        <div class="form-field">
-            @Html.CheckBoxFor(m => m.AlreadyContactedOwner)
-            @Html.LabelFor(m => m.AlreadyContactedOwner, "Yes! I have already tried to contact the package owner about this problem.",
+    </div>
+
+    <!--
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: Why's all this comment out stuff for?

File: src/NuGetGallery/Scripts/gallery/jquery.validate.js (hunk 3 above)
Comment (loic-sharma) [inline src/NuGetGallery/Scripts/gallery/jquery.validate.js]: Yes we do - it looks like the merge didn't resolve this properly. I'll go ahead and fix this.

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml (hunk 4 above)
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: Oops, this is the old HTML - I'll remove this!

File: src/NuGetGallery/ExtensionMethods.cs [hunk 5]
```diff
@@ -300,12 +301,45 @@ public static MailAddress ToMailAddress(this User user)
...
         public static HtmlString ShowTextBoxFor<TModel, TProperty>(this HtmlHelper<TModel> html, Expression<Func<TModel, TProperty>> expression)
         {
+            var htmlAttributes = HtmlAttributesFor(html, expression);
+
+            return html.TextBoxFor(expression, htmlAttributes);
+        }
+
+        public static HtmlString ShowCheckboxFor<TModel>(this HtmlHelper<TModel> html, Expression<Func<TModel, bool>> expression)
+        {
+            var htmlAttributes = HtmlAttributesFor(html, expression, "form-check-input");
```
Comment (joelverhagen) [inline src/NuGetGallery/ExtensionMethods.cs]: What is this class `form-check-input`?

Comment (loic-sharma) [inline src/NuGetGallery/ExtensionMethods.cs]: It's a class for Bootstrap v4, it seems like v3 doesn't support it. I'll remove this.

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml [hunk 6]
```diff
@@ -2,98 +2,97 @@
...
+    <h1 class="page-heading">Report Abuse</h1>
 
-<p class="message">
-    <strong>Important: Please do not use this form to report a bug in a package you are using!</strong><br />
-    This form is for reporting <em>abusive</em> packages such as
-    packages containing <em>malicious code</em> or spam. If "@Model.PackageId" simply doesn't
-    work, or if you need help getting the package installed, please
-    <a href="@Url.Action(actionName: "ContactOwners", controllerName: "Packages", routeValues: new { id = Model.PackageId })" title="contact the owners">contact the owners instead.</a>
-</p>
+    <div class="alert alert-warning" role="alert">
```
Comment (scottbommarito) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: I expect that we'll use this or something similar to it in our other contact forms. We may want to move it into a reusable place.

File: src/NuGetGallery/ExtensionMethods.cs [hunk 7]
```diff
@@ -300,21 +301,56 @@ public static MailAddress ToMailAddress(this User user)
...
+        }
+
+        public static HtmlString ShowCheckboxFor<TModel>(this HtmlHelper<TModel> html, Expression<Func<TModel, bool>> expression)
+        {
+            var htmlAttributes = HtmlAttributesFor(html, expression, addFormControl: false);
+
+            return html.CheckBoxFor(expression, htmlAttributes);
+        }
+
+        public static HtmlString ShowTextareaFor<TModel, TProperty>(this HtmlHelper<TModel> html, Expression<Func<TModel, TProperty>> expression, int rows, int columns)
```
Comment (scottbommarito) [inline src/NuGetGallery/ExtensionMethods.cs]: `ShowTextareaFor` -> `ShowTextAreaFor`

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml [hunk 8]
```diff
@@ -2,98 +2,97 @@
...
+
+    <div class="row report-form">
+        <div class="col-sm-offset-2 col-sm-8">
+            @using (Html.BeginForm())
             {
-                <input type="hidden" name="Email" value="test@example.com" />
+                @Html.AntiForgeryToken()
+                @Html.SpamPreventionFields()
+
+                <div class="form-group @Html.HasErrorFor(m => m.Email)">
```
Comment (scottbommarito) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: Just for the record, I love how simple this code is.

File: src/Bootstrap/less/theme/page-report-abuse.less [hunk 9]
```diff
@@ -0,0 +1,17 @@
+.page-report-abuse {
+  .report-form {
+    margin-top: 40px;
+
+    #alreadycontactedowner-label {
+      display: inline;
+    }
+
+    h2 {
```
Comment (scottbommarito) [inline src/Bootstrap/less/theme/page-report-abuse.less]: IMO this should be more specific
```
.infringement-claim-requirements {
    h2 {
        ...
    }
}
```

Comment (loic-sharma) [inline src/Bootstrap/less/theme/page-report-abuse.less]: This form is unlikely to change, but I don't mind being a little more explicit :)

File: src/NuGetGallery/Views/Packages/ReportAbuse.cshtml (hunk 6 above)
Comment (loic-sharma) [inline src/NuGetGallery/Views/Packages/ReportAbuse.cshtml]: Done. I also moved the package details page over to the new alert helpers.

File: src/NuGetGallery/App_Code/ViewHelpers.cshtml [hunk 10]
```diff
@@ -4,6 +4,33 @@
...
+{
+    @Alert(htmlContent, "info", "Info", isAlertRole)
+}
+
+@helper AlertWarning(Func<MvcHtmlString, HelperResult> htmlContent, bool isAlertRole = false)
+{
+    @Alert(htmlContent, "warning", "Warning", isAlertRole)
+}
+
+@helper AlertDanger(Func<MvcHtmlString, HelperResult> htmlContent, bool isAlertRole = true)
```
Comment (loic-sharma) [inline src/NuGetGallery/App_Code/ViewHelpers.cshtml]: Notice the default for `isAlertRole` is true here.

**Pull request merged**